packages/react-meteor-data/useTracker.js

```diff
@@ -95,38 +92,30 @@ function useTracker(reactiveFn, deps) {
   // if prevDeps or deps are not set areHookInputsEqual always returns false
   // and the reactive functions is always called
   if (!areHookInputsEqual(deps, previousDeps.current)) {
-    dispose();
-
     // Use Tracker.nonreactive in case we are inside a Tracker Computation.
     // This can happen if someone calls `ReactDOM.render` inside a Computation.
     // In that case, we want to opt out of the normal behavior of nested
     // Computations, where if the outer one is invalidated or stopped,
     // it stops the inner one.
-    computation.current = Tracker.nonreactive(() => (
+    Tracker.nonreactive(() => (
       Tracker.autorun((c) => {
+        // This will capture data synchronously on first run (and after deps change).
+        // Additional cycles will follow the normal computation behavior.
+        const data = reactiveFn();
+        if (Meteor.isDevelopment) checkCursor(data);
+        trackerData.current = data;
+
         if (c.firstRun) {
-          const data = reactiveFn();
-          Meteor.isDevelopment && checkCursor(data);
+          // if we are re-creating the computation, we need to stop the old one.
+          dispose();
```

Comment:
Doesn't this cancel reactivity? if the computation is stopped directly on `firstRun` a second run will never happen.

Reply:
The dispose method cancels the one currently in the computation ref:

```js
  const dispose = () => {
    if (computation.current) {
      computation.current.stop();
      computation.current = null;
    }
  }
```

But then we record the new one on the line following:

```js
        if (c.firstRun) {
          // if we are re-creating the computation, we need to stop the old one.
          dispose();

          // store the new computation
          computation.current = c

          // store the deps for comparison on next render
          previousDeps.current = deps;
        } else {
          // use a uniqueCounter to trigger a state change to force a re-render
          forceUpdate(++uniqueCounter);
        }
```

Reply:
I completely missed that, thanks!